architecture/2024-08-resource-types-registration.md

```diff
@@ -0,0 +1,555 @@
+  "properties": {
+    "locations": {
+      "global": {}
+    }
+  }
+}
+```
+
+```json
+{
+  "id": "/planes/radius/local/providers/System.Resources/resourceProviders/Contoso.Platform",
+  "name": "Contoso.Platform",
+  "type": "System.Resources/resourceProviders",
+  "location": "global",
+  "properties": {
+    "locations": {
+      "global": {}
+    },
+    "provisioningState": "Succeeded"
+  },
+  "systemData": {}
+}
+```
+
+**Example: Resource Type PUT**
+
+```txt
+PUT /planes/radius/local/providers/System.Resources/resourceProviders/Contoso.Platform/resourceTypes/contosoBuses
+```
+
+```json
+{
+  "properties": {
+    "apiVersions": {
+      "2024-05-01": {},
+      "2024-08-01": {}
+    },
+    "defaultApiVersion": "2024-08-01",
+    "locations": {
+      "global": {
+        "apiVersions": [
+          "2024-05-01",
+          "2024-08-01"
+        ]
+      }
+    }
+  }
+}
+```
+
+```json
+{
+  "id": "/planes/radius/local/providers/System.Resources/resourceProviders/Contoso.Platform/resourceTypes/contosoBuses",
+  "name": "contosoBuses",
+  "type": "System.Resources/resourceProviders/resourceTypes",
+  "properties": {
+    "apiVersions": {
+      "2024-05-01": {},
+      "2024-08-01": {}
+    },
+    "defaultApiVersion": "2024-08-01",
+    "locations": {
+      "global": {
+        "apiVersions": [
+          "2024-05-01",
+          "2024-08-01"
+        ]
+      }
+    },
+    "provisioningState": "Succeeded"
+  },
+  "systemData": {}
+}
+```
+
+#### Example: Resource Provider Query
+
+```txt
+GET /planes/radius/local/providers
+```
+
+```json
+{
+  "value": [
+    {
+      "name": "Contoso.Platform",
+      "locations": {
+        "global": {},
+        "resourceTypes": {
+          "apiVersions": {
+            "2024-05-01": {},
+            "2024-08-01": {}
+          },
+          "defaultApiVersion": "2024-08-01",
+          "locations": {
+            "global": {
+              "apiVersions": [
+                "2024-05-01",
+                "2024-08-01"
+              ]
+            }
+          },
+        }
+      }
+    }
+  ]
+}
+```
+
+### Implementation Details
+
+#### UCP (if applicable)
+
+##### New APIs
+
+The new APIs will be implemented using our existing *shared* controllers inside the Radius module of UCP. The existing *shared* controllers will be improved as necessary to implement the correct behaviors for a nested resource.
+
+
+##### Bootstrap registration
+
+UCP provides a *bootstrap* implementation that registers planes instances during startup using its configuration file. This will be used to register the resource providers and resource types that Radius includes by default. The *bootstrap* implementation will be enhanced as needed to implement this functionality.
+
+### Error Handling
+
+The important error handling scenarios for this change are related to the nested resource lifecyle.
+
+- Rejecting an operation of a resource type when the parent resource provider resource is processing an operation.
```

Comment:
I'm not sure if this is needed and I've debated taking it out of the doc. Let's discuss.